go/gen.py

```diff
@@ -0,0 +1,33 @@
+import sys
+import pathlib
+import subprocess
+import os
+from python.runfiles import runfiles
+
+try:
+    workspace_dir = pathlib.Path(os.environ['BUILD_WORKSPACE_DIRECTORY'])
+except KeyError:
+    print("this should be run with bazel run", file=sys.stderr)
+    sys.exit(1)
+
+go_extractor_dir = workspace_dir / "go" / "extractor"
+go_dbscheme = workspace_dir / "go" / "ql" / "lib" / "go.dbscheme"
+r = runfiles.Create()
+go, gazelle, go_gen_dbscheme = map(r.Rlocation, sys.argv[1:])
+
+print("updating vendor")
+subprocess.check_call([go, "-C", go_extractor_dir, "work", "vendor"])
+
+print("clearing generated BUILD files")
+for build_file in go_extractor_dir.glob("*/**/BUILD.bazel"):
+    build_file.unlink()
```

Comment:
Question: why is it necessary to delete all the existing `BUILD.bazel` files before running `gazelle`?

Reply:
In general it wouldn't be necessary, but I find it safer:
* when editing `BUILD` files generated by gazelle, gazelle will try to just update them where required
* if we started doing that, the "real" source of truth for the go build would become the `BUILD` files rather than the Go files
* so I prefer keeping the Go files as the only source of truth, which also allows hiding `BUILD` file changes in the github UI (by setting them as generated files, which wouldn't be advisable if we could actually change them :sweat_smile:).

Now that you make think about it, maybe it's better to not delete them locally, but having CI do that to just check at PR level nobody's sneaking in an unwanted change.

Reply:
This makes sense, thank you for explaining that! I didn't realise gazelle would update existing files, rather than overwrite them, so deleting them all to ensure that there's one source of truth explains it. Adding a CI check for this does seem like a nicer solution as well!